docs/framework-support-libraries.md

```diff
@@ -0,0 +1,137 @@
+    }
+]
+ ```
+
+#### LuigiAutoRoutingService
+
+This service cannot be used directly, but it will provide useful features on how to synchronize your Angular application with Luigi navigation.
+It can happen that in your microfrontend, user can navigate through different components/pages.
+With this feature we provide an easy way of synchronizing Angular route with Luigi navigation; in Angular route configuration, you can now add in data these attributes:
+
+ ```javascript
+{path: 'luigi-client-support-preload',component: Sample1Component,data: { fromVirtualTreeRoot: true }}
+{path: 'luigi-client-support-preload',component: Sample2Component,data: { luigiRoute: '/home/sample2' }}
+{path: 'luigi-client-support-preload',component: Sample2Component,data: { luigiRoute: '/home/sample2', fromContext: true}}
+{path: 'luigi-client-support-preload',component: Sample2Component,data: { luigiRoute: '/home/sample2', fromContext: 'localContext'}}
+
+ ```__
+
+with `data: { fromVirtualTreeRoot: true }`, once we load Sample1Component, we will call Luigi Client:
+ ```javascript
+  luigiClient.linkManager().fromVirtualTreeRoot().withoutSync().navigate({route url});
+ ```
+with `data: { luigiRoute: '/home/sample2' }`, uses luigiClient API in this way:
+ ```javascript
+  luigiClient.linkManager().withoutSync().navigate(data.luigiRoute);
+ ```
+with `data: { luigiRoute: '/home/sample2', fromContext: true }`, uses luigiClient API in this way:
+ ```javascript
+  luigiClient.linkManager().fromClosestContext().withoutSync().navigate(data.luigiRoute);
+ ```
+with `data: { luigiRoute: '/home/sample2', fromContext: 'localContext' }`, uses luigiClient API in this way:
+ ```javascript
+  luigiClient.linkManager().fromContext('localContext').withoutSync().navigate(data.luigiRoute);
+ ```
+
+More information about linkManager can be found [here](https://docs.luigi-project.io/docs/luigi-client-api/?section=linkmanager).
+
+
+### LuigiRouteStrategy
+
+To use **LuigiAutoRoutingService**, this library defines a new **RouteReuseStrategy** named **LuigiRouteStrategy**.
+If you need to define your own **RouteReuseStrategy**, you can extend **LuigiRouteStrategy** by overriding it in this way:
+
+ ```javascript
+export class YourRouteStrategy extends LuigiRouteStrategy {
+
+    retrieve(route: ActivatedRouteSnapshot): DetachedRouteHandle {
+        super.retrieve(route);
+        // Your code
+    }
+
+}
+ ```
+and define the provider:
+ ```javascript
+ {
+      provide: RouteReuseStrategy,
+      useClass: YourRouteStrategy
+ }
+ ```
+
+We also provide an example of how to extend **LuigiRouteStrategy** in class **LuigiReuseRouteStrategy**.
+In this class, we added the possibility to "reuse" a component and not initialize it every time you load it (it could be useful to keep component state.)
+
+**LuigiReuseRouteStrategy** can be configured in the following way:
+ ```javascript
+{path: 'luigi-client-support-preload',component: Sample1Component,data: { reuse: true }}
+ ```
```

Comment:
```suggestion
We also provide an example of how to extend **LuigiRouteStrategy** in class **LuigiReuseRouteStrategy**.
In this class, we added the possibility to "reuse" a component and not initialize it every time you load it (it could be useful to keep component state.)

**LuigiReuseRouteStrategy** can be configured in the following way:
 ```javascript
{path: 'luigi-client-support-preload',component: Sample1Component,data: { reuse: true }}
 ```
```

Reply:
If we remove the ``` at the very end, I think the `code block` will not be formatted correctly. (See previous examples in the doc.) All the other text in your suggestion is the same as the original. I'm not sure if you meant to add another suggestion besides removing the ticks 🤔

Reply:
Hi @alexandra-simeonova, 

yes my bad: we actually delete LuigiReuseRouteStrategy from code, so we would just need to remove  from line 131 to line 137. (I fixed the suggestion).

Cheers
Davide